### PR TITLE
trade_ui: address small bugs

### PR DESCRIPTION
**Pull request by andrei8l**
#### Summary
SUMMARY: None

#### Purpose of change
Address small bugs with the new `trade_ui` and remove old workarounds inherited from `trading_window`

Fixes https://github.com/CleverRaven/Cataclysm-DDA/issues/43239: contained liquids have negative prices
Fixes https://github.com/CleverRaven/Cataclysm-DDA/issues/52816: invlet and keybinding conflicts
Fixes https://github.com/CleverRaven/Cataclysm-DDA/issues/52817: allied NPCs still check balance before accepting
Fixes https://github.com/CleverRaven/Cataclysm-DDA/issues/46873: `npc_trading::trade()` can cause a segfault if window refresh is triggered while traded items are manipulated
Can't trade for, or even expand some containers (ex: zipper bag): remove `TRADER_AVOID` from all container items

#### Describe the solution
Please see individual commits. These are all small fixes.

#### Describe alternatives you've considered
N/A

### CODE REVIEW

**Comment by Night-Pryanik** [comment]
Any chance this PR will fix #14619?

**Comment by andrei8l** [comment]
> Any chance this PR will fix #14619?

If I'm reading this chain right, that dialogue option first generates a random debt with parameters defined in json here https://github.com/CleverRaven/Cataclysm-DDA/blob/564d632595faa5feb47c7dc9d34a9a5b932e8c08/data/json/npcs/TALK_COMMON_OTHER.json#L381
then picks an item that can fit in the NPC's total debt, then subtracts the item's value from the debt here
https://github.com/CleverRaven/Cataclysm-DDA/blob/564d632595faa5feb47c7dc9d34a9a5b932e8c08/src/npctalk_funcs.cpp#L577
At the moment I don't see any way to properly remove the fake debt added in the first step, sorry. It probably requires changes to JSON processing so you'd have a single-step effect like  `"effect": [ { "give_equipment": [ ...paremeters here... ] } ]` instead of the two steps used now.

File: src/npc.cpp [hunk 1]
```diff
@@ -2059,8 +2059,7 @@ int npc::value( const item &it ) const
 
 int npc::value( const item &it, int market_price ) const
 {
-    if( it.is_dangerous() || ( it.has_flag( flag_BOMB ) && it.active ) ||
-        it.made_of( phase_id::LIQUID ) ) {
+    if( it.is_dangerous() || ( it.has_flag( flag_BOMB ) && it.active ) ) {
         // NPCs won't be interested in buying active explosives or spilled liquids
```
Comment (wapcaplet) [inline src/npc.cpp:2063]:
```suggestion
        // NPCs won't be interested in buying active explosives
```

**Comment by wapcaplet** [comment]
Nice work! I retested the associated bugs in your branch, and can confirm:

- Liquids (and their containers, and their containers' containers, etc.) have correct positive prices.
- Allied NPCs accept trades that put them in debt (or gifts) without complaint.
- When trading an item inside a wielded backpack for an item that I must wield, I get prompted to drop the backpack, the trade goes through ok, and there is no crash.
- I already changed my keybinding and can't easily test the new default, but `TAB` works to swap panes as expected.

**Comment by andrei8l** [comment]
@wapcaplet thanks a lot for testing and tracking down these bugs!

**Pull request merged**